packages/node_modules/@node-red/nodes/core/network/10-mqtt.js

```diff
@@ -111,19 +111,29 @@ module.exports = function(RED) {
         if (typeof this.cleansession === 'undefined') {
             this.cleansession = true;
         }
-        var prox;
-        if (process.env.http_proxy != null) { prox = process.env.http_proxy; }
-        if (process.env.HTTP_PROXY != null) { prox = process.env.HTTP_PROXY; }
+
+        var prox, noprox;
+        if (process.env.http_proxy) { prox = process.env.http_proxy; }
+        if (process.env.HTTP_PROXY) { prox = process.env.HTTP_PROXY; }
+        if (process.env.no_proxy) { noprox = process.env.no_proxy.split(","); }
+        if (process.env.NO_PROXY) { noprox = process.env.NO_PROXY.split(","); }
+
 
         // Create the URL to pass in to the MQTT.js library
         if (this.brokerurl === "") {
             // if the broker may be ws:// or wss:// or even tcp://
             if (this.broker.indexOf("://") > -1) {
                 this.brokerurl = this.broker;
                 // Only for ws or wss, check if proxy env var for additional configuration
-                if (this.brokerurl.indexOf("wss://") > -1 || this.brokerurl.indexOf("ws://") > -1 )
-                // check if proxy is set in env
-                    if (prox) {
+                if (this.brokerurl.indexOf("wss://") > -1 || this.brokerurl.indexOf("ws://") > -1 ) {
+                    // check if proxy is set in env
+                    var noproxy;
+                    if (noprox) {
+                        for (var i in noprox) {
```

Comment:
Don't use `for-in` loops for arrays.
```suggestion
                        for (var i;i=0;i<noprox.length) {
```

Reply:
haha, just copied if from another place in NR :).

But could I suggest 
for (var i=0; i<noprox.length; i++) {
rather than
for (var i;i=0;i<noprox.length) {
for clarity?

Reply:
Hmm there shouldn't be any for-in loops on arrays in our code... but one may have slipped in somewhere.

And yes, looking at my suggested change again, I wrote garbage core. Your suggestion will actually work, so go with that.

Reply:
ok, new bits in, mqtt and http had the same loop...